modules/heat_conduction/src/actions/ThermalContactBCsAction.C

```diff
@@ -49,7 +49,8 @@ ThermalContactBCsAction::act()
   InputParameters params = _factory.getValidParams(getParam<std::string>("type"));
 
   // Extract global params
-  _app.parser().extractParams(_name, params);
+  if (isParamValid("parser_syntax"))
+    _app.parser().extractParams(getParam<std::string>("parser_syntax"), params);
```

Comment:
That's what I'm saying. These actions currently have to be added by the parser. If they aren't they certainly won't work.